chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -376,27 +386,24 @@ class Bundle extends Aggregate {
         }
       }
     }
-    ArrayBuffer(nameMap.toSeq:_*) sortWith {case ((an, a), (bn, b)) => (a._id > b._id) || ((a eq b) && (an > bn))}
-  }
-  private[chisel3] def toType = {
-    def eltPort(elt: Data): String = {
-      val flipStr: String = if(Data.isFirrtlFlipped(elt)) "flip " else ""
-      s"${flipStr}${elt.getRef.name} : ${elt.toType}"
-    }
-    s"{${namedElts.reverse.map(e => eltPort(e._2)).mkString(", ")}}"
+    ListMap(nameMap.toSeq sortWith { case ((an, a), (bn, b)) => (a._id > b._id) || ((a eq b) && (an > bn)) }: _*)
   }
-  private[chisel3] lazy val flatten = namedElts.flatMap(_._2.flatten)
-  private[chisel3] override def _onModuleClose: Unit = // scalastyle:ignore method.name
-    for ((name, elt) <- namedElts) { elt.setRef(this, _namespace.name(name)) }
 
-  private[chisel3] final def allElements: Seq[Element] = namedElts.flatMap(_._2.allElements)
+  /** Returns a field's contained user-defined Bundle element if it appears to
+    * be one, otherwise returns None.
+    */
+  private def getBundleField(m: java.lang.reflect.Method): Option[Data] = m.invoke(this) match {
+    case d: Data => Some(d)
+    case Some(d: Data) => Some(d)
+    case _ => None
+  }
 
   override def cloneType : this.type = {
```

Comment:
These uses of Record didn't have to be changed to Bundle. (Although, am ambivalent on whichever word.)